### PR TITLE
RC1 - Update System.Formats.Tar docs (Backport #8375)

### PR DESCRIPTION
**Pull request by carlossanlop**
Cherry-picking the commit that was freshly merged to main, and putting it in the temporary rc1 branch. @gewarren

### CODE REVIEW

File: xml/System.Formats.Tar/PaxTarEntry.xml [hunk 1]
```diff
@@ -65,39 +64,33 @@ The entry type of <paramref name="other" /> is not supported in the PAX format.<
...
-
-The following entries are only found in the Extended Attributes dictionary of a PAX entry if certain conditions are met:
-
-- Group name, under the name `gname`, as a string, if it is larger than 32 bytes.
-- User name, under the name `uname`, as a string, if it is larger than 32 bytes.
-- File length, under the name `size`, as an <xref:System.Int32>, if the string representation of the number is larger than 12 bytes.
-
-
-          ]]></format>
+          <para>When creating an instance using the <see cref="M:System.Formats.Tar.PaxTarEntry.#ctor(System.Formats.Tar.TarEntryType,System.String)"/> constructor, only the following entry types are supported: <see cref="F:System.Formats.Tar.TarEntryType.Directory"/>, <see cref="F:System.Formats.Tar.TarEntryType.HardLink"/>, <see cref="F:System.Formats.Tar.TarEntryType.SymbolicLink"/>, <see cref="F:System.Formats.Tar.TarEntryType.RegularFile"/>, <see cref="F:System.Formats.Tar.TarEntryType.BlockDevice"/>, <see cref="F:System.Formats.Tar.TarEntryType.CharacterDevice"/>, and <see cref="F:System.Formats.Tar.TarEntryType.Fifo"/>.</para>
```
Comment (gewarren) [inline xml/System.Formats.Tar/PaxTarEntry.xml:70]: This unfortunately is messed up too: https://review.docs.microsoft.com/en-us/dotnet/api/system.formats.tar.paxtarentry.-ctor?view=net-7.0&branch=main#system-formats-tar-paxtarentry-ctor(system-formats-tar-tarentrytype-system-string)

Comment (carlossanlop) [reply xml/System.Formats.Tar/PaxTarEntry.xml:70]: Ok I'll fix it here first, then later in main. This is the branch from which I need to retrieve the intellisense and the undoc APIs report before tomorrow.

File: xml/System.Formats.Tar/PaxTarEntry.xml [hunk 2]
```diff
@@ -65,39 +64,33 @@ The entry type of <paramref name="other" /> is not supported in the PAX format.<
       <Docs>
         <param name="entryType">The type of the entry.</param>
         <param name="entryName">A string with the path and file name of this entry.</param>
+
```
Comment (carlossanlop) [inline xml/System.Formats.Tar/PaxTarEntry.xml:67]:
```suggestion
```

File: xml/System.Formats.Tar/PaxTarEntry.xml [hunk 3]
```diff
@@ -65,39 +64,33 @@ The entry type of <paramref name="other" /> is not supported in the PAX format.<
...
+          <para>Use the <see cref="M:System.Formats.Tar.PaxTarEntry.#ctor(System.Formats.Tar.TarEntryType,System.String,System.Collections.Generic.IEnumerable{System.Collections.Generic.KeyValuePair{System.String,System.String}})" /> constructor to include additional extended attributes when creating the entry.</para>
+          <para>The following entries are always found in the extended attributes dictionary of any PAX entry:</para>
+          <list type="bullet">
+            <item>Modification time, under the name <c>mtime</c>, as a <see cref="T:System.Double" /> number.</item>
+            <item>Access time, under the name <c>atime</c>, as a <see cref="T:System.Double" /> number.</item>
+            <item>Change time, under the name <c>ctime</c>, as a <see cref="T:System.Double" /> number.</item>
+            <item>Path, under the name <c>path</c>, as a string.</item>
+          </list>
```
Comment (carlossanlop) [inline xml/System.Formats.Tar/PaxTarEntry.xml:78]:
```suggestion
          <format type="text/markdown"><![CDATA[
- Modification time, under the name `mtime`, as a <xref:System.Double> number.
- Access time, under the name `atime`, as a <xref:System.Double> number.
- Change time, under the name `ctime`, as a <xref:System.Double> number.
- Path, under the name `path`, as a string.
          ]]></format>
```

File: xml/System.Formats.Tar/PaxTarEntry.xml [hunk 4]
```diff
@@ -65,39 +64,33 @@ The entry type of <paramref name="other" /> is not supported in the PAX format.<
...
+          <para>The following entries are only found in the extended attributes dictionary of a PAX entry if certain conditions are met:</para>
+          <list type="bullet">
+            <item>Group name, under the name <c>gname</c>, as a string, if it is larger than 32 bytes.</item>
+            <item>User name, under the name <c>uname</c>, as a string, if it is larger than 32 bytes.</item>
+            <item>File length, under the name <c>size</c>, as an <see cref="T:System.Int32" />, if the string representation of the number is larger than 12 bytes.</item>
+          </list>
```
Comment (carlossanlop) [inline xml/System.Formats.Tar/PaxTarEntry.xml:84]:
```suggestion
          <format type="text/markdown"><![CDATA[
- Group name, under the name `gname`, as a string, if it is larger than 32 bytes.
- User name, under the name `uname`, as a string, if it is larger than 32 bytes.
- File length, under the name `size`, as an <xref:System.Int32>, if the string representation of the number is larger than 12 bytes.
          ]]></format>
```

File: xml/System.Formats.Tar/PaxTarEntry.xml [hunk 5]
```diff
@@ -120,41 +113,32 @@ The following entries are only found in the Extended Attributes dictionary of a
...
-
-          ]]></format>
+            <para>Use this constructor to include additional extended attributes when creating the entry.</para>
+            <para>The following entries are always found in the extended attributes dictionary of any PAX entry:</para>
+            <list type="bullet">
+                <item>Modification time, under the name <c>mtime</c>, as a <see cref="T:System.Double" /> number.</item>
+                <item>Access time, under the name <c>atime</c>, as a <see cref="T:System.Double" /> number.</item>
+                <item>Change time, under the name <c>ctime</c>, as a <see cref="T:System.Double" /> number.</item>
+                <item>Path, under the name <c>path</c>, as a string.</item>
+            </list>
```
Comment (carlossanlop) [inline xml/System.Formats.Tar/PaxTarEntry.xml:126]:
```suggestion
            <format type="text/markdown"><![CDATA[
- Modification time, under the name `mtime`, as a <xref:System.Double> number.
- Access time, under the name `atime`, as a <xref:System.Double> number.
- Change time, under the name `ctime`, as a <xref:System.Double> number.
- Path, under the name `path`, as a string.
            ]]></format>
```

File: xml/System.Formats.Tar/PaxTarEntry.xml [hunk 6]
```diff
@@ -120,41 +113,32 @@ The following entries are only found in the Extended Attributes dictionary of a
...
+            <para>The following entries are only found in the extended attributes dictionary of a PAX entry if certain conditions are met:</para>
+            <list type="bullet">
+                <item>Group name, under the name <c>gname</c>, as a string, if it is larger than 32 bytes.</item>
+                <item>User name, under the name <c>uname</c>, as a string, if it is larger than 32 bytes.</item>
+                <item>File length, under the name <c>size</c>, as an <see cref="T:System.Int32" />, if the string representation of the number is larger than 12 bytes.</item>
+            </list>
```
Comment (carlossanlop) [inline xml/System.Formats.Tar/PaxTarEntry.xml:132]:
```suggestion
            <format type="text/markdown"><![CDATA[
- Group name, under the name `gname`, as a string, if it is larger than 32 bytes.
- User name, under the name `uname`, as a string, if it is larger than 32 bytes.
- File length, under the name `size`, as an <xref:System.Int32>, if the string representation of the number is larger than 12 bytes.
            ]]></format>
```

File: xml/System.Formats.Tar/PaxTarEntry.xml [hunk 7]
```diff
@@ -176,27 +160,20 @@ The following entries are only found in the Extended Attributes dictionary of a
...
-
-          ]]></format>
+            <para>The extended attributes are specified when constructing an entry. Use <see cref="M:System.Formats.Tar.PaxTarEntry.#ctor(System.Formats.Tar.TarEntryType,System.String,System.Collections.Generic.IEnumerable{System.Collections.Generic.KeyValuePair{System.String,System.String}})" /> to append your own enumeration of extended attributes to the current entry on top of the default ones. Use <see cref="M:System.Formats.Tar.PaxTarEntry.#ctor(System.Formats.Tar.TarEntryType,System.String)" /> to only use the default extended attributes.</para>
+            <para>The following entries are always found in the extended attributes dictionary of any PAX entry:</para>
+            <list type="bullet">
+                <item>Modification time, under the name <c>mtime</c>, as a <see cref="T:System.Double" /> number.</item>
+                <item>Access time, under the name <c>atime</c>, as a <see cref="T:System.Double" /> number.</item>
+                <item>Change time, under the name <c>ctime</c>, as a <see cref="T:System.Double" /> number.</item>
+                <item>Path, under the name <c>path</c>, as a string.</item>
+            </list>
```
Comment (carlossanlop) [inline xml/System.Formats.Tar/PaxTarEntry.xml:170]:
```suggestion
            <format type="text/markdown"><![CDATA[
- Modification time, under the name `mtime`, as a <xref:System.Double> number.
- Access time, under the name `atime`, as a <xref:System.Double> number.
- Change time, under the name `ctime`, as a <xref:System.Double> number.
- Path, under the name `path`, as a string.
            ]]></format>
```

File: xml/System.Formats.Tar/PaxTarEntry.xml [hunk 8]
```diff
@@ -176,27 +160,20 @@ The following entries are only found in the Extended Attributes dictionary of a
...
+            <para>The following entries are only found in the extended attributes dictionary of a PAX entry if certain conditions are met:</para>
+            <list type="bullet">
+                <item>Group name, under the name <c>gname</c>, as a string, if it is larger than 32 bytes.</item>
+                <item>User name, under the name <c>uname</c>, as a string, if it is larger than 32 bytes.</item>
+                <item>File length, under the name <c>size</c>, as an <see cref="T:System.Int32" />, if the string representation of the number is larger than 12 bytes.</item>
+            </list>
```
Comment (carlossanlop) [inline xml/System.Formats.Tar/PaxTarEntry.xml:176]:
```suggestion
            <format type="text/markdown"><![CDATA[
- Group name, under the name `gname`, as a string, if it is larger than 32 bytes.
- User name, under the name `uname`, as a string, if it is larger than 32 bytes.
- File length, under the name `size`, as an <xref:System.Int32>, if the string representation of the number is larger than 12 bytes.
            ]]></format>
```

File: xml/System.Formats.Tar/PaxTarEntry.xml [hunk 9]
```diff
@@ -67,37 +66,30 @@ The entry type of <paramref name="other" /> is not supported in the PAX format.<
...
 - Modification time, under the name `mtime`, as a <xref:System.Double> number.
 - Access time, under the name `atime`, as a <xref:System.Double> number.
 - Change time, under the name `ctime`, as a <xref:System.Double> number.
 - Path, under the name `path`, as a string.
-
-The following entries are only found in the Extended Attributes dictionary of a PAX entry if certain conditions are met:
-
+          ]]></format>
+          <para>The following entries are only found in the extended attributes dictionary of a PAX entry if certain conditions are met:</para>
+          <format type="text/markdown"><![CDATA[
```
Comment (gewarren) [inline xml/System.Formats.Tar/PaxTarEntry.xml:79]: I would just combine this with the previous CDATA section.

File: xml/System.Formats.Tar/PaxTarEntry.xml [hunk 10]
```diff
@@ -120,41 +112,32 @@ The following entries are only found in the Extended Attributes dictionary of a
...
 - Modification time, under the name `mtime`, as a <xref:System.Double> number.
 - Access time, under the name `atime`, as a <xref:System.Double> number.
 - Change time, under the name `ctime`, as a <xref:System.Double> number.
 - Path, under the name `path`, as a string.
-
-The following entries are only found in the Extended Attributes dictionary of a PAX entry if certain conditions are met:
-
+            ]]></format>
+            <para>The following entries are only found in the extended attributes dictionary of a PAX entry if certain conditions are met:</para>
+            <format type="text/markdown"><![CDATA[
```
Comment (gewarren) [inline xml/System.Formats.Tar/PaxTarEntry.xml:127]: Same here.

File: xml/System.Formats.Tar/PaxTarEntry.xml [hunk 11]
```diff
@@ -176,27 +159,20 @@ The following entries are only found in the Extended Attributes dictionary of a
...
```
Comment (gewarren) [inline xml/System.Formats.Tar/PaxTarEntry.xml:171]: Same.

File: xml/System.Formats.Tar/TarWriter.xml [hunk 12]
```diff
@@ -216,28 +208,30 @@ Before writing an entry to the archive, if you wrote data into the entry's <xref
...
-   - <xref:System.Formats.Tar.TarEntryType.HardLink>
-   - <xref:System.Formats.Tar.TarEntryType.SymbolicLink>
-   - <xref:System.Formats.Tar.TarEntryType.V7RegularFile>
+   - <xref:System.Formats.Tar.TarEntryType.Directory>.
+   - <xref:System.Formats.Tar.TarEntryType.HardLink>.
+   - <xref:System.Formats.Tar.TarEntryType.SymbolicLink>.
+   - <xref:System.Formats.Tar.TarEntryType.V7RegularFile>.
 
-- <xref:System.Formats.Tar.TarEntryFormat.Ustar>, <xref:System.Formats.Tar.TarEntryFormat.Pax>, and <xref:System.Formats.Tar.TarEntryFormat.Gnu>
+- <xref:System.Formats.Tar.TarEntryFormat.Ustar>, <xref:System.Formats.Tar.TarEntryFormat.Pax> and <xref:System.Formats.Tar.TarEntryFormat.Gnu>:
```
Comment (gewarren) [inline xml/System.Formats.Tar/TarWriter.xml]:
```suggestion
- <xref:System.Formats.Tar.TarEntryFormat.Ustar>, <xref:System.Formats.Tar.TarEntryFormat.Pax>, and <xref:System.Formats.Tar.TarEntryFormat.Gnu>:
```

**Review by gewarren — APPROVED** [review]

File: xml/System.Formats.Tar/PaxTarEntry.xml (hunk 9 above)
Comment (gewarren) [reply xml/System.Formats.Tar/PaxTarEntry.xml:79]: Also I think the warnings are indicating that you can't combine XML and HTML in the remarks, so just pick one.

Comment (carlossanlop) [reply xml/System.Formats.Tar/PaxTarEntry.xml]: Merged.

**Pull request merged**